Add mypy doc (#1611)
open-telemetry/opentelemetry.io: content/en/docs/instrumentation/python/mypy.md

---
title: Using mypy

> weight: 5

Reviewer: Should this "note" page be at the end of the Python-section page list?

(If you drop the weight it might migrate to the end of the list.)

---

> If you're using [mypy](http://mypy-lang.org/), you'll need to turn on [namespace
> packages](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-namespace-packages).

Reviewer: ```suggestion
If you're using [mypy](http://mypy-lang.org/), you'll need to turn on [namespace
packages](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-namespace-packages), otherwise `mypy` won't be able to run correctly.
```

> In your project configuration file, add the following:

Reviewer: ```suggestion
To turn on namespace packages, do one of the following:

- Add the following to your project configuration file:
```

If you agree with this change you'll need to indent the toml code block below.

```toml
[tool.mypy]
namespace_packages = true
```

> Or if you'd prefer a command-line switch, use `--namespace-packages`:

Reviewer: ```suggestion
Or, use a command-line switch:
```

If you agree with this change you'll need to indent the code block below.

> ```console

Reviewer: Since you're not adding a leading prompt, you can make the block language `shell` (or add `$` as a prompt).

mypy --namespace-packages
```

If you don't turn on namespace packages, then mypy won't be able to correctly

> run.

Reviewer: Given the changes suggested above you can drop this last sentence.

---
title: Using mypy
weight: 5
---

If you're using [mypy](http://mypy-lang.org/), you'll need to turn on [namespace
packages](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-namespace-packages), otherwise `mypy` won't be able to run correctly.

To turn on namespace packages, do one of the following:

> - Add the following to your project configuration file:

Reviewer 2: Remove the bullet point, or add one below (Or ...)
